### DISCUSSION TITLE
Energy counters wrong after reboot on POWR316D Elite

### DISCUSSION THREAD

**Discussion by paolomeraviglia** [arendst/Tasmota#21085] [Q&A]
Hi All
Device Sonoff POW R316D Elite
Program Version 13.4.0(tasmota32)
Build Date & Time 2024-02-14T16:17:40
Core/SDK Version 2_0_14/4.4.6.240105
Uptime 0T03:05:06
Flash Write Count	167
Boot Count	38

I bought this POWR316D that I would use in the future to control power usage for some devices.
After setting up that with this template, found here https://templates.blakadder.com/sonoff_POWR316D.html

{"NAME":"Sonoff POWR316D","GPIO":[32,0,0,0,0,576,0,0,0,224,9280,0,3104,0,320,0,0,0,0,0,0,9184,9248,9216,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1}

I have this issues:

- on reboot, the device completely lose the energy total power number
- on the main page, the Energy today equals the energy total, even if the device stays up for multiple days.

To solve the first issue, I tried applying a rule to periodically save stuff on the flash:

Rule1 on time#minute do savedata endon

BUT after 4 days I checked with a reboot and the counter dropped from 2.300 to 1.500, and restarted from that.
Is there something wrong with my template?

Consider that I just flashed using the tasmota32.bin to convert the device from factory.
Am I using the wrong firmware?

Except for that, everything seems to work great, and even the metering seems to be kinda precise.

**Comment by sfromis** [comment]
Have you made sure to allow it to have correct local date/time, typically by contacting a NTP server? Without this, time-related logic (including rollover and save at midnight) cannot work properly.

Note that saving data every minute does "cost" in flash write cycles, contributes to wearing down flash in the long run. By default Tasmota will save data at midnight, or when there's other reasons to save data, depending on your selected options - very much including your settings for SaveData and  `SetOption0`.

Not sure what you did to flash, but the recommended procedure is to flash `tasmota32.factory.bin` (or upgrade from another Tasmota binary to `tasmota32.bin`), but with Tasmota operational, this does not look like an issue.

**Reply by paolomeraviglia** [reply]
Thanks for the reply.
I flashed via tasmota32.factory.bin, and I can confirm that the issue is related to timer not set through NTP (because no internet access is allowed / no local NTP server is running in the network).

Unfortunately, my setup requires that the device can run standalone, without networking and when the wifi is found eventually it can connects and sync data to a MQTT broker (and eventually update the time).

I found that the number is now consistent, after changing rule 1 simply setting time on boot to enable timers working:

{"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":39,"Free":472,"Rules":"ON Power1#Boot Do Time 1712091438 ENDON"}}

With this, the energytotal, today and yesterday seems to be set up correctly, but still if I power-cycle the device the counters resets to the last saved state ( that seems to be eg when you update a rule, or the counter saved one hour before ).
Also, I think the previous rule (Rule1 on time#minute do savedata endon) was not working at all (the device had a  Flash Write Count 167, while it was running since 4 days), and I think that was due to missing time sync (so... no time functions at all)

At the end, my main goal is to have a consistent energytotal counter. Do you have any suggestions to reach that goal? Am I using tasmota in an improper way?

**Comment by Jason2866** [comment]
Use command `Restart 2` before removing Power from the device. The commands shuts down Tasmota and save the values. It will start with next re-powering.
